### ISSUE TITLE
Charging the Joy-Cons?

### ISSUE THREAD

**Issue by mturley** [drew-wallace/switchberry#3]
Have you put any time into figuring out how to get the SwitchBerry's joycon rails connected to USB power? I would love to have my SwitchBerry charge my joycons.

I was looking at a few 3rd party joycon charging docks, and many of them only contact the very bottom of the joycon. Underneath that bottom part of the rail is where it makes contact with the charging connector. So, it should be relatively easy to take one of those 3rd party chargers, or the official charging grip, and remix the 3d print file so that the very bottom of the rails are removed, with enough space to attach the parts from the charging dock.  Or, it might be cleaner to remix the file such that instead of the joycon locking into the plastic like that, the plastic is fitted so that we can screw the metal rails from the charging grip onto them. That might make for a more authentic feeling joycon lock, too.

I might find some time soon to take a crack at this. But I wanted to see if you had any ideas.

**Comment by mturley** [comment]
The more I think about this, @drew-wallace, the more I realize both how much I want it and how little time I will have to learn how to mess with 3d printing files.

If you want to try and make this work, **I would be willing to pay you a small bounty** for the 3d print files and guide updates. I'm thinking it might be easiest to go with my second idea above, fitting the metal rails from the joycon charging grip into the sides of the switchberry, instead of joycons sliding into plastic.

**Comment by drew-wallace** [comment]
I have the charging grip and haven't tried taking it apart yet, but I think that would be your best bet. Screwing on the rails might be a little difficult since the screws are tiny, but you could try.

**Comment by mturley** [comment]
Yeah I was thinking that it might be tough to screw those tiny screws right into 3d printed plastic, there is no way you could print threads like that. Maybe a little plate with the threaded hole can sit behind an unthreaded hole, and we can use longer screws. Could just take the threaded side of the screw hole from the guts of the charging grip.

**Comment by mturley** [comment]
I might try a more frankenstein approach at first, and just slice that charging grip right in half and glue the pieces, plastic and all, to the sides of the screen. Then I'll use that to figure out how the charging grip wiring can fit in the box. And make a cleaner one with the rails directly screwed into the print after I get it working.

**Comment by mturley** [comment]
@drew-wallace I am new to 3d printing. What software do you use to edit those files?

**Comment by drew-wallace** [comment]
Believe it or not, the 3D Builder Windows 10 app. It was really easy to use.

**Comment by mturley** [comment]
Ha! Interesting, okay. I'll have to figure something else out because I'm on a mac. But I'll check that out on a windows machine some time, it looks neat.

**Comment by djwraith** [comment]
I created an account specifically to discuss this project!  I wanted to mention that I am building a similar project of my own and intended to do exactly what you're talking about here.  I'll 3d print a case that contains the same measurements as the charging grip for housing the metal bracket in order to incorporate joycon charging.  My issue becomes how to configure joycons to operate as one controller within RetroPie.  I just noticed today that @drew-wallace isn't running RetroPie but instead streaming PC games using Parsec from the Rasbian desktop.  When I get to this point, will it be tricky to pull this off in RetroPie, or do you think your work will translate easily?  I'll give it a shot myself when I get to that point but wondering if you'd be willing to take a crack at it if necessary.  I can open a separate issue if that's cleaner, just thought it pertained to the overall joycon discussion.  Also, I realize you have no obligation to assist  ; )

I greatly appreciate you sharing your work with us!!!

**Comment by djwraith** [comment]
I'm also happy to share my 3d print files when I have them done if it helps anyone else!

**Comment by drew-wallace** [comment]
@djwraith Come on over to the [Discord server](https://discord.gg/kCXVRVK) and chat.
